packages/nectar/src/sagas/publicChannels/checkForMessages/checkForMessages.saga.test.ts

```diff
@@ -1,122 +1,90 @@
 
-  const identity: Identity = {
-    id: 'id',
-    hiddenService: {
-      onionAddress: 'onionAddress.onion',
-      privateKey: 'privateKey'
-    },
-    dmKeys: { publicKey: 'publicKey', privateKey: 'privateKey' },
-    peerId: { id: 'peerId', pubKey: 'pubKey', privKey: 'privKey' },
-    zbayNickname: '',
-    userCsr: undefined,
-    userCertificate: ''
-  }
+    // Set date display format
+    process.env.LC_ALL = 'en_US.UTF-8'
 
-  test.skip('ask for missing messages', async () => {
-    await expectSaga(checkForMessagesSaga)
-      .withReducer(
-        combineReducers({
-          [StoreKeys.PublicChannels]: publicChannelsReducer,
-          [StoreKeys.Communities]: communitiesReducer,
-          [StoreKeys.Identity]: identityReducer
-        }),
-        {
-          [StoreKeys.PublicChannels]: {
-            ...new PublicChannelsState(),
-            channels: communityChannelsAdapter.setAll(
-              communityChannelsAdapter.getInitialState(),
-              [communityChannels]
-            )
-          },
-          [StoreKeys.Communities]: {
-            ...new CommunitiesState(),
-            currentCommunity: 'id',
-            communities: communitiesAdapter.setAll(
-              communitiesAdapter.getInitialState(),
-              [community]
-            )
+    store = prepareStore().store
+
+    factory = await getFactory(store)
+
+    community = await factory.create<
+    ReturnType<typeof communitiesActions.addNewCommunity>['payload']
+    >('Community')
+
+    channel = publicChannels.selectors.publicChannels(store.getState())[0]
+
+    alice = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>(
+      'Identity',
+      { id: community.id, zbayNickname: 'alice' }
+    )
+  })
+
+  test('ask for missing messages', async () => {
+    const types = [MessageType.Basic, MessageType.Empty, MessageType.Empty]
+
+    const shuffled = types
+      .map(value => ({ value, sort: Math.random() }))
+      .sort((a, b) => a.sort - b.sort)
+      .map(({ value }) => value)
+
+    const messages = await Promise.all(
+      shuffled.map(async type => {
+        return await factory.create<
+        ReturnType<typeof publicChannels.actions.test_message>['payload']
+        >('Message', {
+          identity: alice,
+          message: {
+            id: Math.random().toString(36).substr(2.9),
+            type: type,
+            message: 'message',
+            createdAt: DateTime.utc().valueOf(),
+            channelId: 'general',
+            signature: '',
+            pubKey: ''
           },
-          [StoreKeys.Identity]: {
-            ...new IdentityState(),
-            identities: identityAdapter.setAll(
-              identityAdapter.getInitialState(),
-              [identity]
-            )
-          }
-        }
-      )
+          verifyAutomatically: true
+        })
+      })
+    )
+
+    const empties = messages.filter(item => item.message.type === MessageType.Empty)
```

Comment:
Where are empty messages from?

Reply:
We're just not displaying empty messages?

Reply:
We add message object with no data but the id, as we receive the list of ids from waggle. Then we ask for the messages and we update those 'empty' objects with data